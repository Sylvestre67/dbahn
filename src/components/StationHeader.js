import React from 'react';
// import PropTypes from 'prop-types';
import { Query, withApollo } from 'react-apollo';
import { withRouter } from 'react-router-dom';

import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import StationFragments from '../context/fragments';
import { GET_STATION } from '../context/queries';

const styles = theme => ({
	root: {},
});

const StationHeader = props => {
	const { classes, client, match } = props;
	let station;

	try {
		station = client.readFragment({
			id: `Station_${match.params.evaId}`,
			fragment: StationFragments.fragments.station_meta_data,
		});
	} catch (error) {
		station = null;
	}

	return Object.is(station, null) ? (
		<Query
			query={GET_STATION}
			variables={{ evaId: match.params.evaId }}
			client={client}>
			{({ data, loading, error }) => (
				<React.Fragment>
					{loading ? (
						<div />
					) : (
						<div className={classes.root}>
							<Typography variant="h5">
								{data.stationWithEvaId.name}
							</Typography>
							<Typography>
								{JSON.stringify(data.stationWithEvaId, 4, 4)}
							</Typography>
						</div>
					)}
				</React.Fragment>
			)}
		</Query>
	) : (
		<div className={classes.root}>
			<Typography variant="h5">{station.name}</Typography>
			<Typography>{JSON.stringify(station, 4, 4)}</Typography>
		</div>
	);
};

StationHeader.propTypes = {};

StationHeader.defaultProps = {};

export default withStyles(styles)(withApollo(withRouter(StationHeader)));
