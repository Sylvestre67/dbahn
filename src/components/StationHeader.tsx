import React from 'react';
import { ApolloClient, ApolloError } from 'apollo-client';
import { Query, withApollo } from 'react-apollo';
import { withRouter, RouteComponentProps } from 'react-router-dom';

import { withStyles, createStyles, WithStyles } from '@material-ui/core/styles';

import Typography from '@material-ui/core/Typography';

import StationFragments from '../context/fragments';
import { GET_STATION } from '../context/queries';
import { MetaData } from '../context/__generated__/MetaData';
import { Station } from '../context/__generated__/Station';

import StationQuery from './StationQuery';

const styles = () =>
	createStyles({
		root: {},
	});

interface IStationHeaderProps extends WithStyles<typeof styles> {}
interface IStationHeaderProps extends RouteComponentProps<{ evaId: string }> {
	client: ApolloClient<any>;
}

interface IRenderStationHeaderProps extends WithStyles<typeof styles> {
	data: Station | null | undefined;
	loading: boolean;
	error: ApolloError | undefined;
}

const RenderStationHeader: React.SFC<IRenderStationHeaderProps> = props => {
	const { classes, data, loading, error } = props;

	if (loading) {
		return <Typography variant="h5">Loading</Typography>;
	}

	if (data && data.stationWithEvaId) {
		const {
			stationWithEvaId: { name },
		} = data;

		return (
			<div className={classes.root} style={{}}>
				<Typography variant="h5">{name}</Typography>
			</div>
		);
	}

	return <React.Fragment />;
};

const StationHeader: React.SFC<IStationHeaderProps> = props => {
	const { classes, client, match } = props;

	let station_meta_data: MetaData | null;
	try {
		station_meta_data = client.readFragment({
			id: `Station_${match.params.evaId}`,
			fragment: StationFragments.fragments.station_meta_data,
		});
	} catch (error) {
		station_meta_data = null;
	}

	if (station_meta_data) {
		return <Typography variant="h5">{station_meta_data.name}</Typography>;
	} else {
		return (
			<StationQuery
				query={GET_STATION}
				variables={{ evaId: parseInt(match.params.evaId, 10) }}
				client={client}>
				{({ data, loading, error }) => {
					return (
						<RenderStationHeader
							classes={classes}
							data={data}
							loading={loading}
							error={error}
						/>
					);
				}}
			</StationQuery>
		);
	}
};

export default withStyles(styles)(withApollo(withRouter(StationHeader)));
