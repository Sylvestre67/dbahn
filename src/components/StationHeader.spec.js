import React from 'react';
import { shallow } from 'enzyme';

import StationHeader from './StationHeader';
let component;

const props = {
	classes: { root: 'testing' },
};

describe('StationHeader', () => {
	beforeAll(() => {
		component = shallow(<StationHeader {...props} />);
	});

	it('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});
});
