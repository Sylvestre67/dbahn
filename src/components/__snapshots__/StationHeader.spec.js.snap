// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationHeader should render correctly 1`] = `
<withApollo(withRouter(StationHeader))
  classes={
    Object {
      "root": "withApollo-withRouter-StationHeader---root-1 testing",
    }
  }
/>
`;
